### PR TITLE
ppl: add new movements for pin groups to reduce annealing runtime

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

**Comment by maliberty** [comment]
annealing_constraint6 (tcl) *ERROR* SIGABRT
Please cleanup clang-tidy

File: src/ppl/src/Netlist.h [hunk 1]
```diff
@@ -184,7 +184,7 @@ class Netlist
                     bool order,
                     int group_idx);
   void addIOGroup(const std::vector<int>& pin_group, bool order);
-  const std::vector<PinGroupByIndex>& getIOGroups() { return io_groups_; }
+  std::vector<PinGroupByIndex>& getIOGroups() { return io_groups_; }
```
Comment (maliberty) [inline src/ppl/src/Netlist.h:187]: Why is this returning a non-const reference?  It breaks the encapsulation.

Comment (eder-matheus) [reply src/ppl/src/Netlist.h:187]: I overlooked the issue with the encapsulation, but I know how to fix it. My idea was to return a non-const to sort the pin indices according to the edge on which the group is placed. I will create a function in the Netlist class that will perform this sorting to fix the issue.

**Comment by eder-matheus** [comment]
> annealing_constraint6 (tcl) _ERROR_ SIGABRT Please cleanup clang-tidy

I didn't face this when testing on my Mac, but I will check it out in our servers.

**Review by maliberty — APPROVED** [review]

**Pull request merged**